### PR TITLE
Poc/vue override

### PR DESCRIPTION
**Pull request by alaameddeb**

### CODE REVIEW

File: override-sample/src/main/webapp/js/main.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+requirejs(['SHARED/ExtensionRegistry'], function(extensionRegistry) {
+  extensionRegistry.registerComponent('chat', 'exo-modal', ExoModalExtention);
+  extensionRegistry.registerExtension('chat', 'room-actions', newAction);
+});
+
+// define the new action
+const newAction = {
+  key: 'cloneRoom',
+  labelKey: 'Clone the room',
+  class: 'uiIconSocLastestReply'
```
Comment (thomasdelhomenie) [inline override-sample/src/main/webapp/js/main.js]: We should add all the available attributes IMO. I see the attrbute "enabled" is missing. Maybe there are others ?
Also, can we improve the example to define what will happen when clicking on this action ?

File: override-sample/src/main/webapp/js/components/ExoModal.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<template>
```
Comment (thomasdelhomenie) [inline override-sample/src/main/webapp/js/components/ExoModal.vue]: Can we rename the file to ExoModalOverride.vue for example to make it clear it is an override ?

File: application/src/main/webapp/js/lib/extensionRegistry.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+(function(){
+  
+  var register = [];
+  
+  function Module(moduleName) {
+    this.moduleName = moduleName;
+    this.components = [];
+    this.extensions = [];
```
[the remark this answers is not in the document]
Comment (alaameddeb) [reply application/src/main/webapp/js/lib/extensionRegistry.js]: I can not understand what you mean

File: override-sample/src/main/webapp/js/extensionRegistry.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+(function(){
```
[the remark this answers is not in the document]
Comment (alaameddeb) [reply override-sample/src/main/webapp/js/extensionRegistry.js]: This file should be on commons project

File: application/src/main/webapp/js/lib/extensionRegistry.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply application/src/main/webapp/js/lib/extensionRegistry.js]: Forget my comment, I misread the code :stuck_out_tongue_winking_eye:

File: override-sample/src/main/webapp/js/extensionRegistry.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply override-sample/src/main/webapp/js/extensionRegistry.js]: ok :+1:

**Pull request closed without merging**